packages/itwin/grouping-mapping-widget/src/widget/components/hooks/useMappingsOperations.ts

```diff
@@ -30,22 +31,85 @@ const fetchMappings = async (
   }
 };
 
+const fetchExtractionStatus = async (
+  iModelId: string,
+  getAccessToken: GetAccessTokenFn,
+  extractionClient: IExtractionClient,
+  setIconStatus: React.Dispatch<React.SetStateAction<"negative" | "positive" | "warning">>,
+  setIconMessage: React.Dispatch<React.SetStateAction<string>>,
+  setExtractionMessageData: React.Dispatch<React.SetStateAction<ExtractionMessageData[]>>
+) => {
+  setIconStatus("warning");
+  setIconMessage("Extraction status pending.");
+  try {
+    const accessToken = await getAccessToken();
+    const extractions = await extractionClient.getExtractionHistory(accessToken, iModelId, 1);
+    const jobId = extractions[0].jobId;
```

Comment:
When is `fetchExtractionStatus` called? I have an array index out of bounds feeling,

Reply:
Good catch! If the iModel had no extractions this will throw.

Reply:
`fetchExtractionStatus` is called when the component mounts since the `useMappingsOperations.ts` is initialized inside `Mappings.tsx`. I wrapped it in an if that changes `extractionIconData`, but I'm not sure this is an efficient way to deal with this issue.